src/Tizen.NUI/src/internal/WebView/WebContext.cs

```diff
@@ -520,6 +538,23 @@ public void RegisterMimeOverriddenCallback(MimeOverriddenCallback callback)
             if (NDalicPINVOKE.SWIGPendingException.Pending) throw NDalicPINVOKE.SWIGPendingException.Retrieve();
         }
 
+        /// <summary>
+        /// Registers callback for http request interceptor.
+        /// <param name="callback">callback for overriding mime type</param>
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void RegisterHttpRequestInterceptedCallback(HttpRequestInterceptedCallback callback)
+        {
+            httpRequestInterceptedCallback = callback;
+            IntPtr ip = IntPtr.Zero;
+            if (httpRequestInterceptedCallback != null)
+            {
+                ip = Marshal.GetFunctionPointerForDelegate(httpRequestInterceptedProxyCallback);
+            }
+            Interop.WebContext.RegisterRequestInterceptedCallback(SwigCPtr, new HandleRef(this, ip));
```

Comment:
You should store delegate object on somewhere to prevent release by GC
HandleRef could not store object

Reply:
>You must manually keep the delegate from being collected by the garbage collector from managed code. The garbage collector does not track references to unmanaged code.

https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.getfunctionpointerfordelegate?view=net-5.0

>The HandleRef constructor takes two parameters: an Object representing the wrapper, and an IntPtr representing the unmanaged handle. The interop marshaler passes only the handle to unmanaged code, and guarantees that the wrapper (passed as the first parameter to the constructor of the HandleRef) remains alive for the duration of the call.

**remains alive for the duration of the call.**

But you need to hold until WebContext object was alive or callback was set to null

Reply:
Please test with GC
``` c#
GC.Collect();
GC.WaitForPendingFinalizers();
```

Reply:
Oh, my mistake, I did't check `httpRequestInterceptedCallback`